File: lib/prompts/templates/learningTargets.ts
import type { Blueprint, GenerateRequest, SectionSpec } from '../../types'

export function buildPrompt(
  spec: SectionSpec,
  blueprint: Blueprint,
  req: GenerateRequest
): string {
  const { lesson_metadata, curriculum } = req
  return `Generate the learning targets section for this study guide.

## Blueprint Context
- Lesson Title: ${blueprint.title}
- Competency: ${curriculum.competency_code} — ${curriculum.competency_description}
- Learning Targets from Blueprint:
${blueprint.learning_targets.map((lt) => `  ${lt.number}. [${lt.bloom_verb}] ${lt.objective}`).join('\n')}
- Subject: ${lesson_metadata.subject}
- Grade Level: ${lesson_metadata.grade_level}

## Instructions
Generate the learning competency statement and the 3 learning targets using "I can..." format.
Each statement should use the bloom_verb from the blueprint and express what a student will be able to do.
Keep language simple and appropriate for Grade ${lesson_metadata.grade_level}.

## Output Schema
{
  "learning_competency": "string (the overall competency in student-friendly language)",
  "learning_targets": [
    { "number": 1, "statement": "string (I can... format)" },
    { "number": 2, "statement": "string (I can... format)" },
    { "number": 3, "statement": "string (I can... format)" }
  ]
}

Section spec: ${JSON.stringify(spec)}`
}

File: backend/study-guide-agent/app/prompts/templates/learning_targets.py
from __future__ import annotations

from app.types import Blueprint, GenerateRequest


def build_prompt(spec, blueprint: Blueprint, request: GenerateRequest) -> str:
    del spec, request

    learning_targets_schema = """{
    "title": "Learning Targets",
    "competency_focus": {
        "lesson_id": "string",
        "core_concept": "string"
    },
    "targets": [
        {
            "number": 1,
            "bloom_verb": "string",
            "objective": "string",
            "success_look_for": "string"
        }
    ]
}"""

    target_text = "\n".join(
        f"- {target.number}. {target.bloom_verb}: {target.objective}"
        for target in blueprint.learning_targets
    )

    prompt_lines = [
        "Create the learning targets section for a K-12 study guide.",
        "Use the blueprint learning targets directly and keep them student-facing.",
        f"- Lesson title: {blueprint.title}",
        f"- Lesson id: {blueprint.lesson_id}",
        f"- Core concept: {blueprint.core_concept}",
        "- Blueprint learning targets:",
        target_text,
        "Requirements:",
        "- Preserve the same number and order as the blueprint learning targets.",
        "- Keep each objective aligned to the blueprint wording and clearly usable as an 'I can' statement.",
        "- Add a short success_look_for that describes what evidence of learning would look like.",
        "- Keep the section focused on competency and student-facing goals only.",
        "Expected JSON schema:",
        learning_targets_schema,
        "Return only JSON.",
    ]

    return "\n".join(prompt_lines)
